packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,264 @@
+      first: number | null;
+    },
+    filter?: { startDate?: Date; endDate?: Date },
+  ): Promise<{ data: AuditLogType[] }> {
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      throw new GraphQLError('Unauthorized: You are not authorized to perform this action');
+    }
+
+    this.logger.info(
+      'Getting audit logs (organizationId=%s, filter=%o, pagination=%o)',
+      organizationSlug,
+      filter,
+      pagination,
+    );
+    const newCursorId = pagination?.cursorId ? decodeHashBasedCursor(pagination?.cursorId) : null;
+    const lastYear = this.formatToClickhouseDateTime(
+      new Date(new Date().setFullYear(new Date().getFullYear() - 1)).toISOString(),
+    );
+    const limit = sql.raw(String(pagination?.first));
+    const cursorTimestamp = pagination?.cursorTimestamp
+      ? this.formatToClickhouseDateTime(pagination.cursorTimestamp)
+      : lastYear;
+
+    const where: SqlValue[] = [];
+    where.push(sql`organization_id = ${organizationSlug}`);
+
+    if (filter?.startDate && filter?.endDate) {
+      const from = this.formatToClickhouseDateTime(filter.startDate.toISOString());
+      const to = this.formatToClickhouseDateTime(filter.endDate.toISOString());
+      where.push(sql`timestamp >= ${from} AND timestamp <= ${to}`);
+    }
+    if (pagination?.cursorId && pagination?.cursorTimestamp) {
+      where.push(sql`id < ${String(newCursorId?.id)}`, sql`timestamp = ${cursorTimestamp}`);
+    } else if (pagination?.cursorId) {
+      where.push(sql`id < ${String(newCursorId?.id)}`);
+    } else if (pagination?.cursorTimestamp) {
+      where.push(sql`timestamp = ${cursorTimestamp}`);
+    } else {
+      where.push(sql`timestamp <= ${cursorTimestamp}`);
+    }
+
+    const whereClause = where.length > 0 ? sql`WHERE ${sql.join(where, ' AND ')}` : sql``;
+
+    const withoutLimit = sql`
+    SELECT *
+    FROM audit_logs
+    ${whereClause}
+    ORDER BY timestamp DESC, id DESC
+    `;
+    const withLimit = sql`
+    SELECT *
+    FROM audit_logs
+    ${whereClause}
+    ORDER BY timestamp DESC, id DESC
+    LIMIT ${limit}
+    `;
+
+    const query = pagination?.first ? withLimit : withoutLimit;
+    const result = await this.clickHouse.query({
+      query: query,
+      queryId: 'get-audit-logs',
+      timeout: 10000,
+    });
+
+    const data = AuditLogClickhouseArrayModel.parse(result.data);
+
+    return {
+      data,
+    };
+  }
+
+  async exportAndSendEmail(
+    organizationSlug: string,
+    filter: { startDate?: string; endDate?: string },
+  ): Promise<{ ok: { url: string } | null } | { error?: { message: string } | null }> {
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      return {
+        error: {
+          message: 'Unauthorized: You are not authorized to perform this action',
+        },
+        ok: null,
+      };
+    }
```

Comment:
Instead of checking whether the user is the owner of the organization, please use `this.session.assertPerformAction`, and assert for `auditLog:export`.